internal/installercache/installercache_test.go

```diff
@@ -149,8 +191,64 @@ var _ = Describe("installer cache", func() {
 				err := os.WriteFile(fname, []byte("abcde"), 0600)
 				return "", err
 			})
-		_, err := manager.Get(releaseID, releaseMirrorID, "pull-secret", mockRelease, version)
+		l, err := manager.Get(releaseID, releaseMirrorID, "pull-secret", mockRelease, version, clusterId)
 		Expect(err).ShouldNot(HaveOccurred())
+		Expect(l.releaseID).To(Equal(releaseID))
+		Expect(l.clusterID).To(Equal(clusterId))
+		Expect(l.startTime).ShouldNot(BeZero())
+		Expect(l.cached).To(BeFalse())
+		Expect(l.extractDuration).ShouldNot(BeZero())
+		Expect(l.Path).ShouldNot(BeEmpty())
+		testMetricsSentOnRelease(l)
+	})
+
+	It("Should have correct metrics on release object if error getting release URL", func() {
+		releaseID := "4.10-orig"
+		releaseMirrorID := ""
+		version := "4.10.0"
+		clusterId := strfmt.UUID(uuid.NewString())
+		workdir := filepath.Join(cacheDir, "quay.io", "release-dev")
+		fname := filepath.Join(workdir, releaseID)
+
+		mockRelease.EXPECT().GetReleaseBinaryPath(
+			releaseID, gomock.Any(), version).
+			Return(workdir, releaseID, fname, errors.New("Some error!"))
+		l, err := manager.Get(releaseID, releaseMirrorID, "pull-secret", mockRelease, version, clusterId)
+		Expect(err).Should(HaveOccurred())
+		Expect(l.releaseID).To(Equal(releaseID))
+		Expect(l.clusterID).To(Equal(clusterId))
+		Expect(l.startTime).ShouldNot(BeZero())
+		Expect(l.cached).To(BeFalse())
+		Expect(l.extractDuration).Should(BeZero())
+		Expect(l.Path).Should(BeEmpty())
+		testMetricsSentOnRelease(l)
+	})
+
+	It("Should have correct metrics on release object if error extracting release", func() {
```

Comment:
I wouldn't expect this to be a valid case. We'll never call `Release` if `Get` returns an error.

Reply:
I don't see this as an error or an anti pattern as it supports the capture of data for the event and also ensures that we always send an event.

Please see my comment above.

Reply:
Please see comment above.